### PR TITLE
Adjust cardCleaningThreshold for language specific kickoff

### PR DESCRIPTION
**Pull request by amicic**
When transitioning from tracing only state to tracing and card cleaning
state, account that pre-calculated remaining free memory for card
cleaning kickoff threshold should be higher if global kickoff was due to
reasons other than low free memory such a language specific kickoff
(class unloading).

The adjustment for card cleaning kickoff threshold is exactly the same
as the difference between actual free memory at the time of global
kickoff and expected free memory global kickoff threshold.

Without the adjustment, card cleaning and any other states after (most
importantly final STW phase where we reclaim memory) are delayed till
(incorrectly very low) pre-calculated card cleaning threshold is met,
effectively negating any efforts of the early language specific kickoff.

Signed-off-by: Aleksandar Micic <amicic@ca.ibm.com>

### CODE REVIEW

**Comment by amicic** [comment]
@dmitripivkine please, review

File: gc/base/standard/ConcurrentGC.cpp [hunk 1]
```diff
@@ -2528,8 +2530,22 @@ MM_ConcurrentGC::doConcurrentTrace(MM_EnvironmentBase *env,
 		_markingScheme->getWorkPackets()->reuseDeferredPackets(env);
 	}
 
-	/* Switch state if card cleaning stage 1 threshold reached */
-	if( (CONCURRENT_TRACE_ONLY == _stats.getExecutionMode()) && (remainingFree < _stats.getCardCleaningThreshold())) {
+	/* Switch state if card cleaning stage 1 threshold reached
+	   Typically kickoff would be triggered by remainingFree < kickoffThreshold (low heap/tenure occupancy).
+	 * However with some other kickoff criteria (for example, language specific, like class unloading)
+	 * much soonr, so that remainingFree is significantly higher than kickoffThreshold.
```
Comment (dmitripivkine) [inline gc/base/standard/ConcurrentGC.cpp:2536]: typo: sooner

**Review by dmitripivkine — APPROVED** [review]

**Comment by amicic** [comment]
@fjeremic or @0xdaryl please proceed with final review&merge

**Comment by 0xdaryl** [comment]
Jenkins test sanity all

**Comment by 0xdaryl** [comment]
Jenkins build all

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**